lib/liquid/block_body.rb

```diff
@@ -37,7 +36,7 @@ def freeze
 
     private def parse_for_liquid_tag(tokenizer, parse_context)
       while (token = tokenizer.shift)
-        unless token.empty? || token =~ WhitespaceOrNothing
+        unless token.strip.empty?
```

Comment:
This will cause an allocation. I'm not sure the the minimal perf gain justifies this.